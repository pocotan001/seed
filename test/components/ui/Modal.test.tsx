import * as React from "react";
import * as renderer from "react-test-renderer";

jest.mock("focus-trap", () => require("../../mocks/focus-trap"));

const close = () => undefined;

describe("<Modal>", () => {
  it("renders correctly on client", () => {
    jest.doMock("react-dom", () => ({
      createPortal: (node: React.ReactNode) => node
    }));

    jest.doMock("~/config", () => ({
      isServer: false,
      isClient: true
    }));

    const Modal = require("~/components/ui/Modal").default;
    const tree = renderer
      .create(<Modal onRequestClose={close}>alo</Modal>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
