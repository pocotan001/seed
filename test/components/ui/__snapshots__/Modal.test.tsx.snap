// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> should render correctly on client 1`] = `
<div
  aria-modal={true}
  className="sc-htpNat YvQfP"
  role="dialog"
>
  <div
    className="sc-bdVaJa bwMdiZ"
    onClick={[Function]}
    role="presentation"
    tabIndex={-1}
  />
  <div
    className="sc-bwzfXH dsSxso"
  >
    alo
  </div>
</div>
`;
